ksql-engine/src/main/java/io/confluent/ksql/util/KafkaTopicClientImpl.java

```diff
@@ -107,11 +127,62 @@ public void deleteTopics(List<String> topicsToDelete) {
       }
     }
     if (!failList.isEmpty()) {
-      throw new KsqlException("Failed to clean up topics: " + failList.stream().collect(Collectors
-                                                                                          .joining(",")));
+      throw new KsqlException("Failed to clean up topics: " + failList.stream()
+          .collect(Collectors.joining(",")));
+    }
+  }
+
+  @Override
+  public void deleteInternalTopics(String applicationId) {
+    if (!isDeleteTopicEnabled) {
+      log.info("Cannot delete topics since 'delete.topic.enable' is false. ");
+      return;
+    }
+    Set<String> topicNames = listTopicNames();
+    List<String> internalTopics = new ArrayList<>();
+    for (String topicName: topicNames) {
+      if (isInternalTopic(topicName, applicationId)) {
+        internalTopics.add(topicName);
+      }
+    }
+    if (!internalTopics.isEmpty()) {
+      deleteTopics(internalTopics);
     }
   }
 
+  private void init() throws ExecutionException, InterruptedException {
+    DescribeClusterResult describeClusterResult = adminClient.describeCluster();
+    this.nodes = new ArrayList<>(describeClusterResult.nodes().get());
```

Comment:
this can be a local variable

Reply:
Yes, thought of using it in future in other methods too but we can change it if we need it. For now I made it local as you suggested.